cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -311,47 +326,46 @@ func addContainerRule(netLink netlinkwrapper.NetLink, isToContainer bool, addr *
 }
 
 // TeardownPodNetwork cleanup ip rules
-func (os *linuxNetwork) TeardownNS(addr *net.IPNet, table int, log logger.Logger) error {
-	log.Debugf("TeardownNS: addr %s, table %d", addr.String(), table)
-	return tearDownNS(addr, table, os.netLink, log)
+func (os *linuxNetwork) TeardownNS(addrs []*net.IP, table int, log logger.Logger) error {
+	log.Debugf("TeardownNS: addrs %s, table %d", addrs, table)
+	return tearDownNS(addrs, table, os.netLink, log)
 }
 
-func tearDownNS(addr *net.IPNet, table int, netLink netlinkwrapper.NetLink, log logger.Logger) error {
-	if addr == nil {
+func tearDownNS(addrs []*net.IP, table int, netLink netlinkwrapper.NetLink, log logger.Logger) error {
+	if len(addrs) == 0 {
 		return errors.New("can't tear down network namespace with no IP address")
 	}
-	// Remove to-pod rule
-	toContainerRule := netLink.NewRule()
-	toContainerRule.Dst = addr
-	toContainerRule.Priority = toContainerRulePriority
-	err := netLink.RuleDel(toContainerRule)
+	for _, addr := range addrs {
```

Comment:
The `nil` check of `addr` is important. Calling the netLink wrapper with `nil` by accident will silently and without error delete the first `ip rule`. This is what caused all the issues with CNI v1.5.4 last year.

Reply:
I'm even considering switching out the netlink wrapper.. I'll do some tests with https://github.com/mdlayher/netlink

Reply:
.. or a language that has references rather than just pointers :trollface: